packages/data/lib/repositories/firestore_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:domain/repositories/remote_store_repository.dart';
+
+class FirestoreRepository implements RemoteStoreRepository {
+  final FirebaseFirestore _firestore;
+
+  FirestoreRepository(this._firestore);
+
+  @override
+  Future<Map<String, dynamic>?> getDocumentDataByValues(
+    String collectionName,
+    Map<String, dynamic> queryMap,
+  ) async {
+    final collectionRef = _firestore.collection(collectionName);
+
+    Query<Map<String, dynamic>>? query;
+
+    queryMap.forEach((key, value) {
+      query = query == null
+          ? collectionRef.where(
+              key,
+              isEqualTo: value,
+            )
+          : query?.where(
+              key,
+              isEqualTo: value,
+            );
+    });
+
+    final document = await query?.get();
+
+    if (document != null) {
+      return document.docs.isNotEmpty ? document.docs.first.data() : null;
+    }
+
+    return null;
```

Comment:
I think it's better to create here more concrete methods, like checkUserExists, which will receive some UserModel and form the query in repository.

1) it will make our code more readable
2) your solution covers only isEqualTo cases. what if we will want to have a condition with more/less/not and so on?

so, one query - one method in repo seems to be better I think

Reply:
Fixed.